packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -114,21 +115,36 @@ function coerceRef(
     typeof mixedRef !== 'object'
   ) {
     if (__DEV__) {
-      if (returnFiber.mode & StrictMode) {
+      // TODO: Clean this up once we turn on the string ref warning for
+      // everyone, because the strict mode case will no longer be relevant
+      if (returnFiber.mode & StrictMode || warnAboutStringRefs) {
         const componentName = getComponentName(returnFiber.type) || 'Component';
-        if (!didWarnAboutStringRefInStrictMode[componentName]) {
+        if (!didWarnAboutStringRefs[componentName]) {
+          let stringRefWarning;
```

Comment:
can I recommend against constructing the message here like this? This part of the message, which is arguably more critical than the part after, won't be extracted into WARNINGS, and isn't as readable as it could be. Instead, I recommend having 2 calls to warningWithoutStack. Even though the text is duplicated, since they're right next to each other, we can be pretty sure we'll edit both if needed. As an example from another warning PR I'm working on https://github.com/facebook/react/pull/16207/files#diff-63c455475b8566c10993b2daa2c3211bR2535-R2555